tests/test_job.py

```diff
@@ -123,6 +123,92 @@ def test_list(self, connection, add_response, limit, ids, want_params, want_name
 
+    @pytest.mark.parametrize(
+        "limit, ids, status, want_params, want_names",
+        [
+            (
+                1,
+                None,
+                "queued",
+                {"size": "1", "status": "queued"},
+                ["foo"],
+            ),
+            (
+                2,
+                None,
+                None,
+                {"size": "2"},
+                ["foo", "bar"],
+            ),
+            (
+                2,
+                None,
+                "invalid",
+                {"size": "2", "status": "invalid"},
+                [],
+            ),
+            (
+                2,
+                [
+                    "00000000-0000-0000-0000-000000000001",
+                    "00000000-0000-0000-0000-000000000002",
+                    "00000000-0000-0000-0000-000000000003",
+                ],
+                "complete",
+                {
+                    "size": "3",
+                    "id": [
+                        "00000000-0000-0000-0000-000000000001",
+                        "00000000-0000-0000-0000-000000000002",
+                        "00000000-0000-0000-0000-000000000003",
+                    ],
+                    "status": "complete",
+                },
+                ["bar"],
+            ),
+        ],
+    )
+    @responses.activate
+    def test_list_status(
+        self, connection, add_response, limit, ids, status, want_params, want_names
+    ):
+        """Tests that the correct jobs are listed and that the correct query
+        parameters are encoded in the HTTP request to the Xanadu Cloud platform.
+        We ensure that the jobs are filtered by the given status.
+        """
+        data = [
+            {
+                "id": "00000000-0000-0000-0000-000000000001",
+                "name": "foo",
+                "status": "queued",
+                "target": "qpu",
+                "created_at": "2020-01-03T12:34:56.789012+00:00",
+            },
+            {
+                "id": "00000000-0000-0000-0000-000000000002",
+                "name": "bar",
+                "status": "complete",
+                "target": "qpu",
+                "created_at": "2020-01-03T12:34:56.789012+00:00",
+                "finished_at": "2020-01-03T12:34:56.789013+00:00",
+            },
+        ][:limit]
+
+        add_response(body={"data": data}, path="/jobs")
+        if status is not None:
+            have_names = [
+                job.name
+                for job in xcc.Job.list(connection, limit=limit, ids=ids, status=status)
+                if status is not None and job.status == status
+            ]
+        else:
+            have_names = [job.name for job in xcc.Job.list(connection, limit=limit, ids=ids)]
+
+        assert have_names == want_names
+
+        have_params = responses.calls[0].request.params  # pyright: reportGeneralTypeIssues=false
+        assert have_params == want_params
```

Comment:
Given that we already test the job listing functionality above, I would suggest focusing on just the status query parameter. This greatly simplifies the test and gets to the heart of what we really want to verify here:
```suggestion
        add_response(body={"data": []}, path="/jobs")
        xcc.Job.list(connection, limit=limit, status=status)
        
        have_params = responses.calls[0].request.params  # pyright: reportGeneralTypeIssues=false
        have_status_param = have_params.get("status")
        assert have_status_param == want_status_param
```